src/pybind/mgr/cephadm/module.py

```diff
@@ -2141,6 +2141,17 @@ def _apply_service_spec(self, spec: ServiceSpec) -> str:
             raise OrchestratorError('cannot scale %s service below 1' % (
                 spec.service_type))
 
+        if spec.placement.count is not None:
+            host_count = len(self.inventory.keys())
+            if spec.service_type in ['mon', 'mgr']:
+                if spec.placement.count > max(5, host_count):
+                    raise OrchestratorError('Cannot apply %s %s daemons with only %s hosts.' % (
+                        spec.placement.count, spec.service_type, host_count))
+            else:
+                if spec.placement.count > (10 * host_count):
+                    raise OrchestratorError('Cannot apply %s %s daemons with only %s hosts.' % (
```

Comment:
I like more a message like :
f'The  maximum number of {spec.service_type} daemons allowed is {host_count * max_count_per_host}. Increase 'max_count_per_host' cephadm option if it is really needed to have more daemons of this type"